### DISCUSSION TITLE
Convert list data into a related table

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#2286] [Feature Ideas]
Once we support lists as a data type in tables, a useful feature to add would be the ability to convert a list column into a table and vice versa.

---
*If you'd like us to build this feature, please comment below with more details about how you plan to use this functionality.*

**Comment by jeromeperez** [comment]
Hi Feature ++ to create directly table by converting list and export data of table in a usable list for user a bit like an export to spreadsheet format

**Reply by seancolsen** [reply]
Thanks for the feedback, @jeromeperez. I'm interested to better understand what you're describing.

For comparison's sake, I think what @kgodey is describing is a feature that would work like this:

1. You have a `contacts` table with an `emails` column that is a [PostgreSQL array](https://www.postgresql.org/docs/current/arrays.html) of `TEXT` values.

    - `contacts`

        | id | name | emails |
        | -- | -- | -- |
        | 1 | Daffy Duck | `daffy@example.com` `d.duck@example.net` |
        | 2 | Donald Duck | `don@ducks.example.org` |

        _(It's worth noting that Mathesar currently has rudimentary support for arrays. Basically we can display them, but not edit them. Eventually we'll improve this support.)_

1. Then you use this feature to transform that data into two tables that look like this:

    - `contacts`

        | id | name |
        | -- | -- |
        | 1 | Daffy Duck |
        | 2 | Donald Duck |

    - `emails`

        | id | contact | email |
        | -- | -- | -- |
        | 1 | `Daffy Duck` | daffy@example.com |
        | 2 | `Daffy Duck` | d.duck@example.net |
        | 3 | `Donald Duck` | don@ducks.example.org |

The idea here is to help users normalize their data properly.

Is this also what you are talking about? If not, could you provide an example?

**Reply by jeromeperez** [reply]
Exactly
